kimchi/src/tests/keccak.rs

```diff
@@ -0,0 +1,60 @@
+use std::array;
+
+use crate::circuits::{
+    constraints::ConstraintSystem,
+    gate::CircuitGate,
+    polynomials::keccak::{self, ROT_TAB},
+    wires::Wire,
+};
+use ark_ec::AffineCurve;
+use mina_curves::pasta::{Fp, Pallas, Vesta};
+use rand::Rng;
+
+//use super::framework::TestFramework;
+type PallasField = <Pallas as AffineCurve>::BaseField;
+
+fn create_test_constraint_system() -> ConstraintSystem<Fp> {
+    let (mut next_row, mut gates) = { CircuitGate::<Fp>::create_keccak(0) };
+
+    // Temporary workaround for lookup-table/domain-size issue
+    for _ in 0..(1 << 13) {
+        gates.push(CircuitGate::zero(Wire::for_row(next_row)));
+        next_row += 1;
+    }
+
+    ConstraintSystem::create(gates).build().unwrap()
+}
+
+#[test]
+// Test that all of the offsets in the rotation table work fine
+fn test_keccak_table() {
```

Comment:
Please add negative tests so we know the constraints actually trigger

Reply:
yeah, i will add those in the rot test file